underfs/b2/src/main/java/alluxio/underfs/b2/B2UnderFileSystem.java

```diff
@@ -0,0 +1,374 @@
+  }
+
+  @Override
+  protected boolean copyObject(String src, String dst) throws IOException {
+    LOG.debug("copyObject src: {}, dst: {}", src, dst);
+    // source request
+    B2ListFileVersionsRequest srcRequest =
+        B2ListFileVersionsRequest.builder(mBucketId).setStartFileName(src).setPrefix(src).build();
+    B2FileVersion srcFileVersion = null;
+    try {
+      for (B2FileVersion version : mB2StorageClient.fileVersions(srcRequest)) {
+        if (version.getFileName().equals(src)) {
+          srcFileVersion = version;
+        } else {
+          break;
+        }
+      }
+    } catch (B2Exception e) {
+      e.printStackTrace();
+    }
+    if (srcFileVersion == null) {
+      LOG.error("Source object not found: {}", src);
+      return false;
+    }
+    String sourceFileId = srcFileVersion.getFileId();
+    // destination request
+    B2ListBucketsRequest bucketsRequest =
+        B2ListBucketsRequest.builder(mAccountId).setBucketName(dst).build();
+    try {
+      mB2StorageClient.listBuckets(bucketsRequest);
+    } catch (B2Exception e) {
+      e.printStackTrace();
+    }
+    getExistingDirectoryStatus(dst);
+    B2CopyFileRequest copyFileRequest =
+        B2CopyFileRequest.builder(sourceFileId, src).setDestinationBucketId(mBucketId).build();
+    try {
+      mB2StorageClient.copySmallFile(copyFileRequest);
+    } catch (B2Exception e) {
+      e.printStackTrace();
+    }
+
+    return true;
+  }
+
+  @Override
+  public boolean createEmptyObject(String key) {
+    final File mFile = new File(PathUtils
+        .concatPath(CommonUtils.getTmpDir(mUfsConf.getList(PropertyKey.TMP_DIRS, ",")),
+            UUID.randomUUID()));
+    final B2ContentSource source = B2FileContentSource.builder(mFile).build();
+
+    B2UploadFileRequest request =
+        B2UploadFileRequest.builder(mBucketName, key, B2ContentTypes.APPLICATION_OCTET, source)
+            .build();
+
+    try {
+      mB2StorageClient.uploadSmallFile(request);
+      return true;
+    } catch (B2Exception e) {
+      LOG.error("Failed to create an empty object on B2with key:{}", key);
+      return false;
+    }
+  }
+
+  /**
+   * Constructs a new instance of {@link B2UnderFileSystem}.
+   *
+   * @param alluxioURI the {@link AlluxioURI} for this UFS
+   * @param conf       the configuration for this UFS
+   * @return the created {@link B2UnderFileSystem} instance
+   */
+  public static B2UnderFileSystem createInstance(AlluxioURI alluxioURI,
+      UnderFileSystemConfiguration conf) throws ServiceException {
+    LOG.debug("Initializing B2 Storage Client at {}", alluxioURI);
+    B2StorageClient client = B2StorageClientFactory.createDefaultFactory()
+        .create(conf.get(PropertyKey.B2_ACCESS_KEY), conf.get(PropertyKey.B2_SECRET_KEY),
+            USER_AGENT);
+
+    try {
+      B2AccountAuthorization accountAuthorization = client.getAccountAuthorization();
+      LOG.debug("B2AccountAuthorization {}", accountAuthorization);
```

Comment:
Can we combine all the log debugs into a single line that prints out all the desired parameters?